frontend/src/scenes/session-recordings/mobile-replay/androidRecordingPromptBannerLogic.ts

```diff
@@ -0,0 +1,28 @@
+import { connect, kea, path, props, selectors } from 'kea'
+import { versionCheckerLogic, VersionCheckerLogicProps } from 'lib/components/VersionChecker/versionCheckerLogic'
+import posthog from 'posthog-js'
+
+import type { androidRecordingPromptBannerLogicType } from './androidRecordingPromptBannerLogicType'
+
+export const androidRecordingPromptBannerLogic = kea<androidRecordingPromptBannerLogicType>([
+    path(['scenes', 'session-recordings', 'SessionRecordings']),
+
+    props({} as VersionCheckerLogicProps),
+
+    connect((props: VersionCheckerLogicProps) => ({
+        values: [versionCheckerLogic(props), ['usedVersions']],
+    })),
+
+    selectors({
+        shouldPromptUser: [
+            (s) => [s.usedVersions],
```

Comment:
The `used versions` isn't persisted and the `versionCheckerLogic` only loads them in `afterMount` if it wasn't loaded since the `lastCheckTimestamp` which means you will get stuck with:
1. It loads first time and shows the banner
2. Subsequent loading of the banner will not trigger the api call and so will not show

Personally, the _only_ thing I can see really being used by the versionCheckerLogic is the api call and the caching. Other than that it is trying to fit a square peg in a round hole imo.

I think we should just have a dedicated simple hogql query in this logic that leads to a stored `android_version` and a `last_checked` ttl to stop over-fetching. That way it's easy to reason on and doesn't have to conflict with the somewhat different use case of the version banner.

Reply:
🤣 that's where I started but it was 99% the same code... it's quick to split